rcm.html

```diff
@@ -1,118 +1,90 @@
+    'p-i-m-i pattern'                    : 'p-m-i-m pattern ',
+    'p-m-i-m pattern'                    : 'p-i-m-i pattern',
+    'ascending and descending sextuplet' : 'descending and ascending sextuplet',
+    'descending and ascending sextuplet' : 'ascending and descending sextuplet',
+    'p-a dyad-m-i triplet pattern'       : 'p-i dyad-m-a triplet pattern',
+    'p-i dyad-m-a triplet pattern'       : 'p-a dyad-m-i triplet pattern',
+    'p-m dyad-i-a triplet pattern'       : 'p-i dyad-a-m triplet pattern',
+    'p-i dyad-a-m triplet pattern'       : 'p-m dyad-i-a triplet pattern ',
+  };
+  const arpeggio1 = getRandomElement(Object.keys(arpeggios));
+  const arpeggio2 = arpeggios[arpeggio1];
+  const arpeggioSpeed = arpeggio1.includes('sextuplet') ? '52' : '76';
 
   // the pieces I am required to memorize
-  var piece = get_random_element(
-    ["Bach Minuet",
-    "Paganini Minuet",
-    "Spanish Romance"]);
+  const piece = getRandomElement([
+    'Bach Minuet',
+    'Paganini Minuet',
+    'Spanish Romance'
+  ]);
 
   // the studies I am required to memorize
-  var study = get_random_element(
-    ["Carcassi Tremolo Study",
-    "Bosch Lamento"]);
+  const study = getRandomElement([
+    'Carcassi Tremolo Study',
+    'Bosch Lamento'
+  ]);
 
   // output!
-  scale_output    = "<b>Scales: </b>" + scale1 + " and " + scale2 + ". " + note_value + ", " + stroke + ", " + fingering;
-  if (tremolando)
-  {
-    scale_output = scale_output + ". Tremolando";	
-  }
-  arpeggio_output = "<b>Arpeggio: </b>" + arpeggio1 + " and " + arpeggio2 + " at " + arpeggio_speed;
-  piece_output    = "<b>Piece: </b>" + piece;
-  study_output    = "<b>Study: </b>" + study;
+  const scaleOutput = `<strong>Scales: </strong> ${scale1} and ${scale2}. ${noteValue}, ${stroke}, ${fingering}${tremolando ? '. Tremolando' : ''}`;
+  const arpeggioOutput = `<strong>Arpeggio: </strong> ${arpeggio1} and ${arpeggio2} at ${arpeggioSpeed}`;
+  const pieceOutput = `<strong>Piece: </strong> ${piece}`;
+  const studyOutput = `<strong>Study: </strong> ${study}`;
 
-  return "<p>" + scale_output    + "</p>" +
-         "<p>" + arpeggio_output + "</p>" +
-         "<p>" + piece_output    + "</p>" +
-         "<p>" + study_output    + "</p>";
-}
+  return '<p>' + [scaleOutput, arpeggioOutput, pieceOutput, studyOutput].join('</p><p>') + '</p>';
+};
 
-cached_practice = localStorage.getItem("cached_practice")
-cached_practice_date = localStorage.getItem("cached_practice_date")
 
-var today = new Date().getDate();
+const cachedPractice = localStorage.getItem('cached_practice');
+const cachedPracticeDate = localStorage.getItem('cached_practice_date');
+const today = String(new Date().getDate());
 
 // if we can use a cached practice, then do so
-if (cached_practice != null &&
-    cached_practice_date == today)
-{
-  document.getElementById("result").innerHTML = cached_practice;
+if (cachedPractice !== null && cachedPracticeDate === today) {
```

Comment:
triple === is generally better than double == for equality checks because it is stricter and checks type as well as value, and does not depend on sometimes confusing type coercion behavior.  This is why I had to String() the `today` variable above, because `.getDate()` returns a number